.github/workflows/sonar.yaml

```diff
@@ -0,0 +1,33 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Sonar Report
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        go: [ '1.21', '1.22', '1' ]
```

Comment:
Think we don't need to run each Go version for this one, since we only need it generating the report once 👍

Reply:
great point